site/content/en/docs/tasks/run_flux.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "Run A Flux Minicluster"
+date: 2022-02-14
+weight: 6
+description: >
+  Run a Kueue scheduled Flux minicluster.
+---
+
+This page shows how to leverage Kueue's scheduling and resource management capabilities when running [Flux Operator's](https://flux-framework.org/flux-operator/) Mini Clusters.
+
+For more information on Flux Operator visit [Flux Operator](https://flux-framework.org/flux-operator/).
+
+
+## Before you begin
+
+Please check [Run A Job](/docs/tasks/run_jobs) for details on the initial cluster setup and queue configuration.
+
+## 1. MiniCluster definition
+
+Since a Flux MiniCluster uses  a `batch/job` to run the actual workload, no extra components are needed for it to be managed by Kueue, However the following aspects should be taken into account:
+
+### a. Queue selection
+
+The target local queue should be specified in the `spec.jobLabels` section of the minicluster configuration.
+
+```yaml
+  jobLabels:
+    kueue.x-k8s.io/queue-name: user-queue
+```
+
+### b. Configure the resource needs
+
+The resource needs of the workload can be configured in the `spec.container[].resources` sections of the minicluster configuration.
```

Comment:
If the user is running something with MPI or similar and they want more control, they also might want to consider tweaking `spec.container[].cores`.  or `spec.tasks` https://flux-framework.org/flux-operator/getting_started/custom-resource-definition.html?h=tasks#tasks. In layman's terms these are telling the exact resources that should be used for the single command. As an example from our kubecon experiment run, on a size 64 cluster:

```yaml
apiVersion: flux-framework.org/v1alpha1
kind: MiniCluster

metadata:
  name: lammps
  namespace: flux-operator
spec:
  # localDeploy needs to be false
  localDeploy: false

  # Number of pods to create for MiniCluster
  size: 64
  tasks: 6016

  # don't suppress output (in case there is issue) and time flux
  logging:
    quiet: false
    timed: true
    debug: true

  # This is a list because a pod can support multiple containers
  containers:
    - image: ghcr.io/rse-ops/spack-ubuntu-libfabric-ssh:ubuntu-20.04
      workingDir: /home/flux/examples/reaxff/HNS
      command: lmp -v x 64 -v y 16 -v z 16 -in in.reaxc.hns -nocite

      # Option Flags for this flux runner wait.sh entrypoint
      fluxOptionFlags: "-ompi=openmpi@5 -c 1 -o cpu-affinity=per-task" 

      # Leave 2 cores for kubernetes, we have 96,
      # this is just for eksctl
      cores: 96

      # Resource limits to enable efa
      resources:
        limits:
          vpc.amazonaws.com/efa: 1
          memory: "340G"
          cpu: 94

        requests:
          vpc.amazonaws.com/efa: 1
          memory: "340G"
          cpu: 94

      # custom preCommand logic (run at start of script)
      preCommand: |
        source /etc/profile.d/z10_spack_environment.sh
        asFlux="sudo -u flux -E PYTHONPATH=$PYTHONPATH -E PATH=$PATH -E FI_EFA_USE_DEVICE_RDMA=1 -E RDMAV_FORK_SAFE=1"
        . /etc/profile.d/z10_spack_environment.sh 
        cd /opt/spack-environment
        . /opt/spack-environment/spack/share/spack/setup-env.sh
        spack env activate .
        cd /home/flux/examples/reaxff/HNS
```
That generated the following command for flux start:

```bash
🌀 flux start $brokerOptions flux mini submit  -N 64 -n 6016 --quiet -ompi=openmpi@5 -c 1 -o cpu-affinity=per-task --watch -vvv lmp -v x 64 -v y 16 -v z 16 -in in.reaxc.hns -nocite
```
Note that `-n` is 6016 and `-N` is 64, and in flux this maps to:

```bash
  -N, --nodes=N               Number of nodes to allocate
  -n, --ntasks=N              Number of tasks to start
```

Reply:
For the purpose of queuing, we care about how many resources the pods will consume in the cluster.
We will obtain those from the created Job object.